### PR TITLE
Feature/dask batch processing

### PR DESCRIPTION
**Pull request by benjijamorris**
- Resolves [https://github.com/AllenInstitute/aics-segmentation/issues/5], adds command line option to dask parallelize segmentation on per directory mode
- Fixes the template segmentation workflow

Rough benchmarking on the template segmentation workflow shows ~6x speedup on per_dir segmentation

### CODE REVIEW

**Comment by jxchen01** [comment]
is `dask` in the dependency?

File: aicssegmentation/bin/batch_processing.py [hunk 1]
```diff
@@ -252,70 +312,26 @@ def execute(self, args):
...
-                    SegModule(
-                        struct_img,
-                        args.mitotic_stage,
-                        self.rescale_ratio,
-                        args.output_type,
-                        output_path,
-                        fn,
-                    )
+            if args.dask:
+                # using dask offers ~6x speedup on the template segmentation.
```
Comment (jxchen01) [inline aicssegmentation/bin/batch_processing.py:322]: what is the computing evironment used for benchmarking? how many CPUs?

**Review by jxchen01 — CHANGES_REQUESTED** [review]
add `dask` to setup.py

File: aicssegmentation/bin/batch_processing.py (hunk 1 above)
Comment (benjijamorris) [reply aicssegmentation/bin/batch_processing.py:322]: This was on windows with 4 CPUs

File: aicssegmentation/bin/batch_processing.py [hunk 2]
```diff
@@ -252,70 +312,26 @@ def execute(self, args):
...
+                import dask
+
+                lazy_results = []
```
Comment (sandraoluoch) [inline aicssegmentation/bin/batch_processing.py:325]: Just for clarification, what is the purpose of "lazy_results" and what is being stored in there?

File: setup.py [hunk 3]
```diff
@@ -72,9 +73,7 @@
     ],
     description="Part 1 of Allen Cell and Structure Segmenter",
     entry_points={
-        "console_scripts": [
-            "my_example=aicssegmentation.bin.my_example:main"
-        ],
+        "console_scripts": ["my_example=aicssegmentation.bin.my_example:main"],
```
Comment (jxchen01) [inline setup.py:76]: i think this step should be  `batch_processing=aicssegmentation.bin.batch_processing:main`

here is another example: https://github.com/AllenCell/segmenter_model_zoo/blob/9cfe7e2882b29ec03409a9957eb6f27e90bbd3bc/setup.py#L72

Comment (jxchen01) [reply setup.py:76]: `my_example` is the dummy function in original cookie-cutter

**Review by jxchen01 — CHANGES_REQUESTED** [review]

File: aicssegmentation/bin/batch_processing.py (hunk 2 above)
Comment (benjijamorris) [reply aicssegmentation/bin/batch_processing.py:325]: My understanding is that it holds the setup for all the calculations, and then when you call dask.compute it does the actual computation that's been set up. 

Here's a link to the dask parallelization tutorial: https://examples.dask.org/applications/embarrassingly-parallel.html

File: setup.py (hunk 3 above)
Comment (benjijamorris) [reply setup.py:76]: Good catch!

**Pull request merged**